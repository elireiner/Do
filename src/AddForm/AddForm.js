import React, { Component } from 'react';

export default class AddForm extends Component{

    render(){
        return(
            <form onSubmit={this.props.handleSubmit}>
                <label>
                    Add to do:
                    <input name="add"/>
                </label>
                <button type="submit">Add Do To</button>
            </form>
        )
    }
}