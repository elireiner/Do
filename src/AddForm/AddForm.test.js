import React from 'react';
import AddForm from './AddForm';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'

describe('AddForm', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<AddFrom />, div);
        ReactDOM.unmountComponentAtNode(div)
    })
    it('renderer UI as expected', () => {
        const tree = renderer.create(<AddForm/>).toJSON();
        expect(tree).toMatchSnapShot()
    })
    it('calls handleSubmit when button is clicked', () => {
        const mockFunction = jest.fn()
        const wrapper = shallow(<AddForm handleSubmit={mockFunction}/>);
        wrapper.find('button').at('one').simulate('click');
        expect(mockFunction).toHaveBeenCalled();
    })
})