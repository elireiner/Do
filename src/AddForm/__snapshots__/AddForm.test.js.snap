// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddForm renderer UI as expected 1`] = `
<form>
  <label>
    Add to do:
    <input
      name="add"
    />
  </label>
  <button
    type="submit"
  >
    Add Do To
  </button>
</form>
`;
